tests/Properties.hs

```diff
@@ -1240,22 +1240,36 @@ prop_initsBB xs = inits xs == map P.unpack (P.inits (P.pack xs))
 
 prop_tailsBB xs = tails xs == map P.unpack (P.tails (P.pack xs))
 
-prop_findSubstringsBB :: String -> Int -> Int -> Bool
+-- The correspondence between the test 'ByteString' and naive test 'String'
+-- must be injective, otherwise the ByteString may find matches at positions
+-- that don't match in the "corresponding" string.  To that end, we start
+-- with and pack a Word8 array, rather than a unicode String.
+--
+prop_findSubstringsBB :: [Word8] -> Int -> Int -> Bool
```

Comment:
Could you possibly use [`AsciiString`](http://hackage.haskell.org/package/QuickCheck-2.14/docs/Test-QuickCheck-Modifiers.html#t:ASCIIString) here?